desktop/core/src/desktop/js/reactComponents/FileChooser/PathBrowser/PathBrowser.tsx

```diff
@@ -0,0 +1,113 @@
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useRef, useEffect, useState, RefObject } from 'react';
+import { Breadcrumb, Input } from 'antd';
+
+import HdfsIcon from '../../../components/icons/HdfsIcon';
+import S3Icon from '../../../components/icons/S3Icon';
+import AdlsIcon from '../../../components/icons/AdlsIcon';
+
+import { BreadcrumbData } from '../types';
+import './PathBrowser.scss';
+
+interface PathBrowserProps {
+  breadcrumbs?: BreadcrumbData[];
+  onFilePathChange: (path: string) => void;
+}
+
+const PathBrowser: React.FC<PathBrowserProps> = ({ breadcrumbs, onFilePathChange }) => {
+  const [inputFieldDisplay, setInputFieldDisplay] = useState(false);
+
+  const icons = {
+    //hdfs file system begins with the first breadcrumb as "" (ex: /user/demo)
+    '': <HdfsIcon />,
+    abfs: <AdlsIcon />,
+    s3: <S3Icon />
+  };
+
+  const useOutsideAlerter = (ref: RefObject<HTMLDivElement>) => {
+    useEffect(() => {
+      // Alert if clicked on outside of element
+      const handleClickOutside = (event: MouseEvent) => {
+        const current = ref?.current;
+        if (current && !current.contains(event.target as Node)) {
+          setInputFieldDisplay(false);
+        }
+      };
+      // Bind the event listener
+      document.addEventListener('mousedown', handleClickOutside);
+      return () => {
+        // Unbind the event listener on clean up
+        document.removeEventListener('mousedown', handleClickOutside);
+      };
+    }, [ref]);
+  };
+
+  const onEnteredCustomPath = e => {
+    onFilePathChange(e.target.value);
+  };
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  useOutsideAlerter(wrapperRef);
+
+  if (breadcrumbs) {
+    return (
+      <div className="path-browser" onClick={e => e.stopPropagation()}>
+        {/* TODO: handle the case when there is a deep file tree or file/directory name is long (too many breadcrumbs cannot be displayed on the modal) */}
+        {!inputFieldDisplay ? (
+          <>
+            {/* .substring() is used to remove //: in the case of s3:// or  abfs:// and so on in all the occurances*/}
+            <div className="filesystem__icon">
+              {icons[breadcrumbs[0].label.substring(0, breadcrumbs[0].label.length - 3)]}
+            </div>
+            <Breadcrumb className="path-browser__breadcrumb" separator=">">
+              {breadcrumbs.map((item: BreadcrumbData, index: number) => {
+                return (
+                  <Breadcrumb.Item
+                    key={index}
```

Comment:
Nit: normally we should avoid using index as key in react due to performance reasons but these lists won't grow that long so I don't think it is a problem here.

Reply:
a unique id of the item itself is such exists

Reply:
Not only performance but also potential bugs when items are replaced which would result in the same key being reused, https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/. 

A potential key choice in this case could be the complete path up to that point if readily available.

Reply:
Yes, good clarification. The order of these items do not change so I think we would be fine using index here, but you can use the item.url here which should be unique for all breadcrumbs.